pkg/prometheus/promcfg.go

```diff
@@ -4124,6 +4146,27 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	return cfg, nil
 }
 
+func (cg *ConfigGenerator) generateOTLPConfig(_ *assets.StoreBuilder) (yaml.MapItem, error) {
+	if cg.version.LT(semver.MustParse("2.54.0")) {
+		return yaml.MapItem{}, fmt.Errorf("OTLP configuration is only supported from Prometheus version 2.54.0")
+	}
+
+	cfg := yaml.MapSlice{}
+	otlpConfig := cg.prom.GetCommonPrometheusFields().OTLP
+
+	if len(otlpConfig.PromoteResourceAttributes) > 0 {
```

Comment:
we don't really need to check for length.

Reply:
Thanks. Got it. Done.